hooks/gen_docs.py

```diff
@@ -1,10 +1,263 @@
+KPOPS_COMPONENTS_INHERITANCE_REF: dict = {
+    component.get_component_type(): component.__base__.get_component_type()  # type: ignore[reportGeneralTypeIssues]
+    for component in KPOPS_COMPONENTS
+}
+KPOPS_COMPONENTS_FIELDS = {
+    # `set` to make it pickleable
+    component.get_component_type(): set(component.__fields__.keys())
+    for component in KPOPS_COMPONENTS
+}
+SECTIONS_ORDER = [
+    "type",
+    "name",
+    "namespace",
+    "app",
+    "from_",
+    "to",
+    "prefix",
+    "repo_config",
+    "version",
+    "resetter_values",
+    "offset_topic",
+]
+
+#####################
+# EXAMPLES          #
+#####################
+
+
+def filter_sections(
+    component_name: str, sections: list[str], include_inherited: bool = False
+) -> list[str]:
+    """Find all sections that are specific to a component from a list
+
+    :param component_name: Component name
+    :param sections: Available section files, names with extension, no path
+    :param include_inherited: Whether to include sections that can be defined in
+        a parent component defaults
+    :returns: The applicable sections
+    """
+    component_sections: list[str] = []
+    for target_section in KPOPS_COMPONENTS_FIELDS[component_name]:
+        component = component_name
+        section = target_section + "-" + component + ".yaml"
+        if section in sections:
+            component_sections.append(section)
+        elif (
+            not include_inherited
+            and KPOPS_COMPONENTS_INHERITANCE_REF[component] == "pipeline-component"
+        ):
+            section = target_section + ".yaml"
+            if section in sections:
+                component_sections.append(section)
+        elif include_inherited:
+            while component != "pipeline-component":
+                component = KPOPS_COMPONENTS_INHERITANCE_REF[component]
+                section = target_section + "-" + component + ".yaml"
+                if section in sections:
+                    component_sections.append(section)
+                    break
+                elif component == "pipeline-component":
+                    section = target_section + ".yaml"
+                    if section in sections:
+                        component_sections.append(section)
+                        break
+    component_sections.sort(key=component_section_position_in_definition)
+    return component_sections
+
+
+def concatenate_text_files(*file_paths: Path, target: Path) -> None:
+    """Concatenate the given files into one
+
+    :param *file_paths: Files to be conatenated. The order of the inputs will be
+        retained in the result
+    :param target: File to store the result in. Note that any previous contents
+        will be deleted.
+    """
+    res = ""
+    for file in file_paths:
+        with open(file, "r") as f:
+            res += f.read()
+    with open(target, "w+") as f:
+        f.write(res)
```

Comment:
I would prefer skipping the intermediate string assignment. you could write directly to the file in append mode
something like this 
```suggestion
    with target.open("a") as f:
        for source in file_paths:
            f.write(source.read_text())
```

Reply:
Good idea! I will have to erase the contents of `target` before the loop then.